rio_tiler/mosaic/reader.py

```diff
@@ -41,6 +43,76 @@ def _filter_tasks(tasks: TaskType):
             pass
 
 
+@dataclass
+class MosaicReader:
```

Comment:
I don't think this is possible right now, there is too much collision between the base classes, but it would be nice for `MosaicReader` to extend `cogeo_mosaic.backends.BaseBackend`.  This way the user can naively pass a mosaicjson into `MosaicReader` and read a tile without having to worry about asset selection.

Reply:
> Maybe this is better off in another lib, I can see the argument where rio-tiler should only be concerned with reading tiles from image(s).

Yes I will tend to agree with that.